server/container/guice/mailbox/src/main/java/org/apache/james/modules/mailbox/MailboxListenersLoaderImpl.java

```diff
@@ -74,18 +76,44 @@ public void register(MailboxListener listener) {
 
     @Override
     public MailboxListener createListener(ListenerConfiguration configuration) {
-        String listenerClass = configuration.getClazz();
-        try {
+        return OptionalUtils
+            .orSuppliers(
+                () -> createFromClass(configuration),
+                () -> createFromFactory(configuration))
+            .orElseThrow(() -> new RuntimeException("unexpected error loading mailbox listener"));
```

Comment:
It's not un unexpected error: you need to provide a clear error message to the user!

Like `MailboxListener needs to specify either 'class' of 'factoryClass'`

Reply:
No, this case is handled in ListenerConfiguration. This error is for the case where ListenerConfiguration and this class diverge with time

Reply:
Sorry, I missed that point, I was thinking that it was a client facing error.